drivers/dma/dma_intel_adsp_gpdma.c

```diff
@@ -107,21 +107,13 @@ static int intel_adsp_gpdma_config(const struct device *dev, uint32_t channel,
 		return res;
 	}
 
-	struct dma_block_config *block_cfg = cfg->head_block;
-
 	/* Assume all scatter/gathers are for the same device? */
 	switch (cfg->channel_direction) {
 	case MEMORY_TO_PERIPHERAL:
-		LOG_DBG("%s: dma %s configuring llp for destination %x",
-			__func__, dev->name, block_cfg->dest_address);
-		intel_adsp_gpdma_llp_config(dev, channel,
-					    block_cfg->dest_address);
-		break;
 	case PERIPHERAL_TO_MEMORY:
-		LOG_DBG("%s: dma %s configuring llp for source %x",
-			__func__, dev->name, block_cfg->source_address);
-		intel_adsp_gpdma_llp_config(dev, channel,
-					    block_cfg->source_address);
+		LOG_DBG("%s: dma %s configuring llp for %x",
+			__func__, dev->name, cfg->dma_slot);
+		intel_adsp_gpdma_llp_config(dev, channel, cfg->dma_slot);
```

Comment:
If instead of address its a dma peripheral id/handshake value then intel_adsp_gpdma_llp_config() which has addr as a param should also be updated slightly to change the variable name there to avoid confusion